### DISCUSSION TITLE
M5 Core2

### DISCUSSION THREAD

**Discussion by KerbalFoundries** [sle118/squeezelite-esp32#272] [Show and tell]
Following some issues I had with a fake WROVER, I found I have an M5stack Core2 which uses a genuine D0WDQ6-V3 chip. It satisfies the memory and flash requirements, soooo.... Flashed with the GUI tool and YES! It works.

It's a neat little module and includes a DAC, amplifier and speaker (model=i2s,bck=12,ws=0,do=02) if anyone is interested in the NVS to get this working for testing. I'll move on to working with the 1334A DAC as the inbuilt stuff isn't up to much.

The core 2 includes a really nice little 2.0"@320*240 screen driven by an ILI9342C. I haven't even looked into whether Squeezelite will drive that yet, but I'll keep the thread updated.

**Comment by sle118** [comment]
You should post some pictures here. Thank you for sharing!

**Comment by KerbalFoundries** [comment]
Nice little units. The stackable dev boards are rather handy too.

The one challenge I've got with the screen is it's powered via an LDO output from an AXP192 power management unit. I believe this needs to be fed some commands via I2C to enable the LD02 line to power the screen. I'm not seeing anything that jumps out in NVS to give that options, and I've not had a good look over the code yet. Any pointers most welcome, though I've aware I'm outside normal use-case territory 😃 


![2023-07-20 07 16 08](https://github.com/sle118/squeezelite-esp32/assets/7767858/a81d3343-3b96-4dad-844b-25ce65d0b76f)
![2023-07-20 07 15 50](https://github.com/sle118/squeezelite-esp32/assets/7767858/d313da88-7b30-4de6-b110-50a3bce5b698)
![2023-07-20 07 16 40](https://github.com/sle118/squeezelite-esp32/assets/7767858/36fbd0b2-80fa-4880-957d-06215c8e7697)

**Reply by sle118** [reply]
There isn't a direct way to do this in NVS. However, the notr sequence has a nod to the old DOS and you can add as many autoexec commands that fit in three digits (e.g. autoexec2, etc). From there, if I remember correctly, you can send i2c commands. It's a long shot, but worth a try if you want to skip the cofrfee change.

**Reply by KerbalFoundries** [reply]
Thanks for the hints. Autoexecs are working great for sending i2cset commands... 

I'm part-way there. I can get the screen working if I flash something the inits the AXP192 correctly to power up the LCD and control logic, then reflash SQ Lite and apply NVS without resetting the module. That way I know I've got solid SPI and display params and I've had the display working. The AXP192 retains all its registers while it's powered, of course, so the challenge is figuring out all the stuff needed for a solid initialisation of the LCD and control electronics.

I'm wading through some M5 code to see how M5.begin() handles everything for this module, then replicating that using i2cget and i2cset as best I can. I'm clearly missing something still, as it's eluded me thus far. I've got the i2c stuff nicely sorted now as i2cdetect returns results and I can toggle the backlight on and off via registers, which is at least a visible change.

**Comment by philippe44** [comment]
You can also use a fake DAC i2c command by dac_controlset
